blueprints/cloud-operations/network-dashboard/cloud-function/main.py

```diff
@@ -17,20 +17,55 @@
 from code import interact
 from distutils.command.config import config
 import os
-from pickletools import int4
 import time
 from google.cloud import monitoring_v3, asset_v1
 from google.protobuf import field_mask_pb2
 from googleapiclient import discovery
-from metrics import ilb_fwrules, instances, networks, metrics, limits, peerings, routes
+from metrics import ilb_fwrules, instances, networks, metrics, limits, peerings, routes, subnets
+
+
+def get_monitored_projects_list(config):
+  '''
+    Gets the projects to be monitored from the MONITORED_FOLDERS_LIST environment variable.
+    
+      Parameters:
+        config (dict): The dict containing config like clients and limits
+      Returns:
+        monitored_projects (List of strings): Full list of projects to be monitored
+    '''
+  monitored_projects = config["monitored_projects"]
+  monitored_folders = os.environ.get("MONITORED_FOLDERS_LIST").split(",")
+
+  # Gets all projects under each monitored folder (and even in sub folders)
+  for folder in monitored_folders:
+    read_mask = field_mask_pb2.FieldMask()
+    read_mask.FromJsonString('name,versionedResources')
+
+    response = config["clients"]["asset_client"].search_all_resources(
+        request={
+            "scope": f"folders/{folder}",
+            "asset_types": ["cloudresourcemanager.googleapis.com/Project"],
+            "read_mask": read_mask
+        })
```

Comment:
Additional finding:
this throws a weird error when setting `monitored_folders_list = []` in the tfvars:
`details = "Invalid scope [folders/]. Please provide a valid scope: projects/{PROJECT_ID}, projects/{PROJECT_NUMBER}, folders/{FOLDER_NUMBER} or organizations/{ORGANIZATION_NUMBER}."`

Trying to debug atm

Reply:
Update: seems if the env is empty there is one empty string in the list.

Reply:
I added a test for this and it should not crash anymore.